beacon-chain/casper/sharding_test.go

```diff
@@ -116,11 +120,136 @@ func TestSampleAttestersAndProposers(t *testing.T) {
 		t.Errorf("Get attesters failed, expected: %v got: %v", validatorList[:len(attesters)], attesters)
 	}
 
-	indices, err = ValidatorsByHeightShard(common.Hash{'A'}, validators, 1, 0)
+	indices, err := ShuffleValidatorsToCommittees(common.Hash{'A'}, validators, 1, 0)
 	if err != nil {
 		t.Errorf("validatorsByHeightShard failed with %v:", err)
 	}
-	if len(indices) != 8192 {
-		t.Errorf("incorret length for validator indices. Want: 8192. Got: %v", len(indices))
+	if len(indices) != params.CycleLength {
+		t.Errorf("incorret length for validator indices. Want: %d. Got: %d", params.CycleLength, len(indices))
+	}
+}
+
+func TestGetCommitteeParamsSmallValidatorSet(t *testing.T) {
+	numValidators := params.CycleLength * params.MinCommiteeSize / 4
+
+	committesPerSlot, slotsPerCommittee := getCommitteeParams(numValidators)
+	if committesPerSlot != 1 {
+		t.Fatalf("Expected committeesPerSlot to equal %d: got %d", 1, committesPerSlot)
+	}
+
+	if slotsPerCommittee != 4 {
+		t.Fatalf("Expected slotsPerCommittee to equal %d: got %d", 4, slotsPerCommittee)
+	}
+}
+
+func TestGetCommitteeParamsRegularValidatorSet(t *testing.T) {
+	numValidators := params.CycleLength * params.MinCommiteeSize
+
+	committesPerSlot, slotsPerCommittee := getCommitteeParams(numValidators)
+	if committesPerSlot != 1 {
+		t.Fatalf("Expected committeesPerSlot to equal %d: got %d", 1, committesPerSlot)
+	}
+
+	if slotsPerCommittee != 1 {
+		t.Fatalf("Expected slotsPerCommittee to equal %d: got %d", 1, slotsPerCommittee)
+	}
+}
+
+func TestGetCommitteeParamsLargeValidatorSet(t *testing.T) {
+	numValidators := params.CycleLength * params.MinCommiteeSize * 8
+
+	committesPerSlot, slotsPerCommittee := getCommitteeParams(numValidators)
+	if committesPerSlot != 5 {
+		t.Fatalf("Expected committeesPerSlot to equal %d: got %d", 5, committesPerSlot)
+	}
+
+	if slotsPerCommittee != 1 {
+		t.Fatalf("Expected slotsPerCommittee to equal %d: got %d", 1, slotsPerCommittee)
+	}
+}
+
+func TestValidatorsByHeightShardRegularValidatorSet(t *testing.T) {
+	validatorIndices := []uint32{}
+	numValidators := params.CycleLength * params.MinCommiteeSize
+	for i := 0; i < numValidators; i++ {
+		validatorIndices = append(validatorIndices, uint32(i))
+	}
+
+	shardAndCommitteeArray := splitBySlotShard(validatorIndices, 0)
+
+	if len(shardAndCommitteeArray) != params.CycleLength {
+		t.Fatalf("Expected length %d: got %d", params.CycleLength, len(shardAndCommitteeArray))
+	}
+
+	for i := 0; i < len(shardAndCommitteeArray); i++ {
+		shardAndCommittees := shardAndCommitteeArray[i].ArrayShardAndCommittee
+		if len(shardAndCommittees) != 1 {
+			t.Fatalf("Expected %d committee per slot: got %d", params.MinCommiteeSize, 1)
```

Comment:
In this case, since line 190 would throw nil ptr exception then fatal is appropriate. 

However, the fatal on 181 unnecessarily ends the test. 

Consider all of my Errorf comments as optional and to apply my suggestions only where it makes sense. I think this might be too opinionated so I don't think enforcing Errorf always is a requirement.